### PR TITLE
Implement mail alias admin system.

### PR DESCRIPTION
**Pull request by jforberg**
I went and did this

### CODE REVIEW

**Comment by jforberg** [comment]
After this is merged:

1. Copy data from postfix.alias table to this new table in the rails database. Easy to do in interactive mode.
2. Change the alias rule in postfix so it fetches from the new place.
3. Never have to handle another email alias again

File: app/models/mail_alias.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        return MailAlias.none
+      end
+
+      # The queries below only work if targets != []
+
+      to_remove = base_scope.where 'target not in (?)', targets
+      to_remove.delete_all # Not validated
+
+      existing = base_scope.where 'target in (?)', targets
+      existing.update_all :updated_at => DateTime.now # Not validated
```
Comment (davidwessman) [inline app/models/mail_alias.rb]: I'd prefer `Time.zone.now`here.

File: app/assets/stylesheets/style/mail_aliases.sass [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#alias-tbody
```
Comment (davidwessman) [inline app/assets/stylesheets/style/mail_aliases.sass]: We are using `.scss` on the other stylesheets.

**Comment by davidwessman** [comment]
Looking really good! Nice and sleek integration.
Not sure how to configure postfix to work with this, but it is probably manageable.
Could we limit the postfix user to only have access to read this table?

**Comment by jforberg** [comment]
Configuring postfix is very easy. Yes, we can limit his access which I think is a good idea. 

If you search for an alias that doesn't exist, you will be given the option to create it. Creating is done in exactly the same way as updating. 

To delete an alias you just update it to point to zero addresses. 

I think this is a simpler design. My goal is to avoid writing views which I find very tedious :)

**Comment by davidwessman** [comment]
I can see that it becomes less view code. But I would prefer to use the somewhat similar setups.

The search is marvelous with JS, but I do not know enough JS to simply understand and edit the javascript part of the functions. 

It seems inevitable to do some JS and I should probably learn. 
I think this solution looks smooth, but I am still a bit reserved.

**Comment by jforberg** [comment]
Ok. I understand your concern. My gui code is quite ugly, like most gui code tends to be.

Still, server side html templating is a bit old fashioned and it's possible to do much nicer UX design with ajax, in some cases. Like how you can edit each row of the table and feedback is instant. That makes it worth it in my opinion.

I'm not a fan of javascript but it just sucks less than alternatives ERB and full HTTP reloads for every action.

If you have questions about my code I'd be happy to answer. If you don't know javascript this could be an opportunity to learn.

**Comment by jforberg** [comment]
ERB and full HTTP reloads can make sense for some applications which fit the HTML "document" model better. Items like posts and pages and photos fit into documents quite neatly.

But for something like this, where it's obvious that there are no documents but just a simple GUI application to configure email aliases, the doucment model just gets in the way so I remove it. The user doesn't care about documents, they just want to edit some aliases like they do in excel, google sheets etc.

**Comment by jforberg** [comment]
@davidwessman Do you object to merging this? I don't want to impose a maintenance burden on you but at the same time "I do not know enough JS" is a very solvable problem :)

**Comment by davidwessman** [comment]
No 😄 
I would like to see Hound a bit more pleased - but go ahead 😄

**Comment by jforberg** [comment]
Well, the hound only has two complaints: I use the more general hash syntax consistently and I put spaces inside compound literals consistently.

This is not an error and is part of my normal code style. I prefer this style because it is more consistent.

If you would prefer me to code in some different way I will happily consider it. But I won't change my style because a computer program tells me to.

**Comment by davidwessman** [comment]
Do as you like.

**Comment by jforberg** [comment]
System is live, migration complete.

**Comment by davidwessman** [comment]
Great work! 😄

**Pull request merged**